util/src/main.ts

```diff
@@ -45,44 +45,56 @@ export function main(): void {
 	.usage('Usage: $0 [validate|visualize] [file] --inputFormat=[line|json] [filters]')
 
 	// Validation tool
-	.command('validate [file]', '', (argv: yargs.Argv) => argv
-		.positional('file', {
+	.command('validate [file]', '',
+	{
+		file: {
 			describe: 'Path to input file or --stdin',
-		}) as any /* ToDo@jumattos */,  (argv: yargs.Arguments<{ stdin: boolean; file: string; inputFormat: string }>) => {
-			if (!argv.stdin && argv.file === undefined) {
-				yargs.showHelp('log');
-				console.error('\nError: Missing input file. Did you forget --stdin?');
-				process.exitCode = 1;
-			} else {
-				readInput(argv.inputFormat, argv.stdin ? '--stdin' : argv.file, (input: LSIF.Element[]) => {
+			type: 'string',
+		},
+	}, (argv: yargs.Arguments) => {
+		if (!argv.stdin && argv.file === undefined) {
+			yargs.showHelp('log');
+			console.error('\nError: Missing input file. Did you forget --stdin?');
+			process.exitCode = 1;
+		} else {
+			readInput(
+				argv.inputFormat as string,
+				argv.stdin ? '--stdin' : argv.file as string,
+				(input: LSIF.Element[]) => {
 					const filter: IFilter = argv as unknown as IFilter;
-					process.exitCode = validate(input, getFilteredIds(filter, input),
-												path.join(path.dirname(process.argv[1]),
-														'../node_modules/lsif-protocol/lib/protocol.d.ts'));
+					process.exitCode = validate(
+						input, getFilteredIds(filter, input),
```

Comment:
Would suggest placing each argument for the `validate` function on a new line so it's easier to read.
With all the nested functions and callbacks, it's already difficult to read.